spec/requests/application_spec.rb

```diff
@@ -1,8 +1,8 @@
 require 'rails_helper'
 
-RSpec.describe 'Application', type: :request do
-  it 'renders a 404 if active record model is not found' do
-    get user_path(-1)
-    expect(response).to have_http_status :not_found
-  end
-end
+# RSpec.describe 'Application', type: :request do
```

Comment:
What's wrong with this?

Reply:
There is no longer any scenario in which the 404 page is shown because the relevant controllers show custom warnings instead. See https://github.com/hpi-swt2/connections-portal/pull/120#discussion_r559004973

Reply:
Maybe the logic could then be tested with another controller that does not have custom warnings yet (e.g. the `NotesController`)

Reply:
The `NotesController` redirects unauthorized requests.

Reply:
If you removed all usages of this logic, I think the logic itself and the tests for it could be removed too. But I still don't really like to build this ourself in every controller :sweat_smile:

Reply:
But it allows us to provide more convenient behavior. I would not like to remove this convenience again ...

But I would also not delete the logic because it does no harm but will provide a pretty good fallback for any new controller that does not implement error handling itself.

Reply:
I agree that this logic provides convenience but after you removed all usages of it, it is pretty much useless. But fine for me.